packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -49,192 +42,236 @@ describeCompat("GC trailing ops tests", "NoCompat", (getTestObjectProvider) => {
 			},
 			loaderProps: { configProvider },
 		};
-		const testContainerConfig: ITestContainerConfig = tombstoneEnabled
-			? tombstoneConfig
-			: defaultGCConfig;
 
 		/**
-		 * Submits a summary and returns the unreferenced timestamp for all the nodes in the container. If a node is
-		 * referenced, the unreferenced timestamp is undefined.
-		 * @returns a map of nodeId to its unreferenced timestamp.
+		 * Validates that the data store is not deleted from the data store and GC trees in the summary.
+		 * Also, it is referenced / unreferenced as expected.
 		 */
-		async function getUnreferencedTimestamps(summaryTree: ISummaryTree) {
+		function validateDataStoreStateInSummary(
+			summaryTree: ISummaryTree,
+			dataStoreNodePath: string,
+			expectGCStateHandle: boolean,
+			referenced: boolean,
+		) {
+			// Validate that the data store is not deleted from the data store summary tree.
+			const dataStoreId = dataStoreNodePath.split("/")[1];
+			const channelsTree = (summaryTree.tree[channelsTreeName] as ISummaryTree).tree;
+			assert.equal(
+				Object.keys(channelsTree).includes(dataStoreId),
+				true,
+				`Data store ${dataStoreId} should not have been deleted from the summary`,
+			);
+
+			// If expecting the GC state to be a handle, validate that and return.
+			if (expectGCStateHandle) {
+				assert.equal(
+					summaryTree.tree[gcTreeKey].type,
+					SummaryType.Handle,
+					"Expecting the GC tree to be handle",
+				);
+				return;
+			}
+
+			// Validate that the GC state does contains an entry for the data store.
 			const gcState = getGCStateFromSummary(summaryTree);
 			assert(gcState !== undefined, "GC tree is not available in the summary");
-			const nodeTimestamps: Map<string, number | undefined> = new Map();
-			for (const [nodeId, nodeData] of Object.entries(gcState.gcNodes)) {
-				nodeTimestamps.set(nodeId.slice(1), nodeData.unreferencedTimestampMs);
-			}
-			return nodeTimestamps;
+			const dataStoreGCData = gcState.gcNodes[dataStoreNodePath];
+			assert(
+				dataStoreGCData !== undefined,
+				`Data store ${dataStoreNodePath} should be present in GC state`,
+			);
+			assert.equal(
+				dataStoreGCData.unreferencedTimestampMs ? false : true,
+				referenced,
+				`Data store ${dataStoreNodePath}'s referenced state is incorrect`,
+			);
+
+			// Validate that the deleted nodes in the GC data does not have the data store.
+			const deletedNodesState = getGCDeletedStateFromSummary(summaryTree);
+			assert.equal(
+				deletedNodesState?.includes(dataStoreNodePath) ?? false,
+				false,
+				`Data store ${dataStoreNodePath} should not be in deleted nodes`,
+			);
 		}
 
 		beforeEach(async function () {
 			provider = getTestObjectProvider({ syncSummarizer: true });
+			if (provider.driver.type !== "local") {
+				this.skip();
+			}
+			settings = {};
```

Comment:
This may not be working as expected. The mockConfigProvider needs to be re-initialized at the beginning of every test.